### ISSUE TITLE
Add a "coverage badge"

### ISSUE THREAD

**Issue by brynrhodes** [cqframework/cql-tests-runner#6]
Support repositories ability to advertise coverage using a badge that shows coverage percentage

**Comment by preston** [comment]
Submitted via PR #75 , which depends on #74.

**Comment by brynrhodes** [comment]
The badges at the top of the readme are great, but they are about this repository. This issue was more about making badges available to engine repositories to state their level of coverage with respect to these tests. I'm not sure a) what that looks like technically, nor b) whether anyone would actually do that?

**Comment by preston** [comment]
Ah ok I see what you're saying. Technically we can already do this with HL7's "Shields" server using "Dynamic JSON Badges". The way it works is you use a normal Markdown image link to the Shields server with a data file URL, JSON path string, and other query params and then the Shields server dynamically creates the badge for the client. The only thing that's missing is a "percentage passing" statistic to query for, as I'm not sure this can be done dynamically. For example:

![Dynamic JSON Badge](https://shields.foundry.hl7.org/badge/dynamic/json?url=https%3A%2F%2Fcql-tests-results.quality.hl7.org%2F202604021352_results.json&query=testResultsSummary.passCount&suffix=%20passing&label=CQL%20Tests&color=green)

**Comment by preston** [comment]
Here's the documentation and badge builder page: https://shields.foundry.hl7.org/badges/dynamic-json-badge

**Comment by brynrhodes** [comment]
![Dynamic JSON Badge](https://shields.foundry.hl7.org/badge/dynamic/json?url=https%3A%2F%2Fcql-tests-results.quality.hl7.org%2Flatest-cqf-java-results.json&query=testResultsSummary.passCount&suffix=%20passing&label=CQL%20Tests&color=green)

**Comment by brynrhodes** [comment]
Is there some refresh that needs to happen in order for new results to be available at cql-tests-results.quality.hl7.org?

**Comment by preston** [comment]
Yes .. The deployment is automatic, but the build is manual only because the `hl7-bot` account doesn't have privileges to install the CI system build script hook. I'd prefer going automated :)

**Comment by preston** [comment]
i just pushed that build for you .. it's live now: https://cql-tests-results.quality.hl7.org/index.json

**Comment by brynrhodes** [comment]
Fantastic, just added it to the java engine project, as soon as that PR merges we'll get to see one live, then I'll recommend that other engines add badges to their repos as well.

**Comment by preston** [comment]
If you want to create a ticket for "percentage" statistics in the results schema, I'll get to it when I can.
